src/main/java/com/frostwire/jlibtorrent/SessionManager.java

```diff
@@ -1119,6 +1119,10 @@ public void run() {
         t.start();
     }
 
+    public TorrentStats trackTorrentStats(TorrentHandle torrentHandle, long samplingIntervalInMs, long maxHistoryInMs) {
+        return new TorrentStats(this, torrentHandle, samplingIntervalInMs, maxHistoryInMs);
```

Comment:
now let's fix this, we don't want the user of this method to be burdened with managing these object instances, let's try to reduce the chance for fuck up.

What we want is another class to keep track of these instances, or like we were talking about, what we want is to have a single `AlertListener` that decides wether to process the incoming alert based on the `torrentHandle`'s we want being tracked.

So you can go about this by adding a new class, but what I'd do instead, would be to make the `AlertListener` you are creating a bit more powerful, and have it keep track of the `TorrentStats` objects it's supposed to maintain or not.